pkg/globalmanager/lifelonglearningjob.go

```diff
@@ -0,0 +1,776 @@
+	jc.enqueueByPod(pod, true)
+}
+
+// obj could be an *sedna.LifelongLearningJob, or a DeletionFinalStateUnknown marker item,
+// immediate tells the controller to update the status right away, and should
+// happen ONLY when there was a successful pod run.
+func (jc *LifelongLearningJobController) enqueueController(obj interface{}, immediate bool) {
+	key, err := k8scontroller.KeyFunc(obj)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", obj, err))
+		return
+	}
+
+	backoff := time.Duration(0)
+	if !immediate {
+		backoff = getBackoff(jc.queue, key)
+	}
+
+	jc.queue.AddAfter(key, backoff)
+}
+
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+// It enforces that the syncHandler is never invoked concurrently with the same key.
+func (jc *LifelongLearningJobController) worker() {
+	for jc.processNextWorkItem() {
+	}
+}
+
+func (jc *LifelongLearningJobController) processNextWorkItem() bool {
+	key, quit := jc.queue.Get()
+	if quit {
+		return false
+	}
+	defer jc.queue.Done(key)
+
+	forget, err := jc.sync(key.(string))
+	if err == nil {
+		if forget {
+			jc.queue.Forget(key)
+		}
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("Error syncing lifelonglearning job: %v", err))
+	jc.queue.AddRateLimited(key)
+
+	return true
+}
+
+// sync will sync the lifelonglearning job with the given key if it has had its expectations fulfilled, meaning
+// it did not expect to see any more of its pods created or deleted. This function is not meant to be invoked
+// concurrently with the same key.
+func (jc *LifelongLearningJobController) sync(key string) (bool, error) {
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).Infof("Finished syncing lifelonglearning job %q (%v)", key, time.Since(startTime))
+	}()
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return false, err
+	}
+	if len(ns) == 0 || len(name) == 0 {
+		return false, fmt.Errorf("invalid lifelonglearning job key %q: either namespace or name is missing", key)
+	}
+	sharedLifelongLearningJob, err := jc.jobLister.LifelongLearningJobs(ns).Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("lifelonglearning job has been deleted: %v", key)
+			return true, nil
+		}
+		return false, err
+	}
+	lifelonglearningjob := *sharedLifelongLearningJob
+	// set kind for lifelonglearningjob in case that the kind is None
+	lifelonglearningjob.SetGroupVersionKind(sednav1.SchemeGroupVersion.WithKind("LifelongLearningJob"))
+
+	// lifelonglearningjob first start
+	if lifelonglearningjob.Status.StartTime == nil {
+		now := metav1.Now()
+		lifelonglearningjob.Status.StartTime = &now
+	}
+
+	// if lifelonglearningjob was finished previously, we don't want to redo the termination
+	if IsLifelongLearningJobFinished(&lifelonglearningjob) {
+		return true, nil
+	}
+
+	forget := false
+	jobFailed := false
+	needUpdated := false
+
+	// update conditions of incremental job
```

Comment:
the comment is about incremental-job still

Reply:
fix it